src/workerd/io/worker-interface.c++

```diff
@@ -34,12 +34,14 @@ public:
   }
 
   kj::Promise<void> connect(kj::StringPtr host, const kj::HttpHeaders& headers,
-      kj::AsyncIoStream& connection, ConnectResponse& response) override {
+      kj::AsyncIoStream& connection, ConnectResponse& response,
+      kj::HttpConnectSettings settings) override {
     KJ_IF_MAYBE(w, worker) {
-      return w->get()->connect(host, headers, connection, response);
+      return w->get()->connect(host, headers, connection, response, kj::mv(settings));
     } else {
-      return promise.addBranch().then([this, host, &headers, &connection, &response]() {
-        return KJ_ASSERT_NONNULL(worker)->connect(host, headers, connection, response);
+      return promise.addBranch().then([this, host, &headers, &connection, &response, settings]() {
+        return KJ_ASSERT_NONNULL(worker)->connect(
+            host, headers, connection, response, kj::mv(settings));
```

Comment:
Hmm, normally I'd expect this to require a `mutable `lambda -- I guess this compiles because `settings`'s move constructor doesn't currently require a mutable object.